components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/impl/BrandingPreferenceApiServiceImpl.java

```diff
@@ -107,6 +142,31 @@ public Response deleteBrandingPreference(String type, String name, String locale
         return Response.noContent().build();
     }
 
+    @Override
+    public Response deleteCustomText(String type, String name, String locale, String screen) {
+
+        if (type != null) {
+            if (!(ORGANIZATION_TYPE.equals(type) || APPLICATION_TYPE.equals(type) || CUSTOM_TYPE.equals(type))) {
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            if (!ORGANIZATION_TYPE.equals(type)) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+        } else {
+            type = ORGANIZATION_TYPE;
+        }
```

Comment:
Shall we make this logic a bit clearer?

```suggestion
        if (type == null) {
            type = ORGANIZATION_TYPE;
        } else if (!(ORGANIZATION_TYPE.equals(type) || APPLICATION_TYPE.equals(type) || CUSTOM_TYPE.equals(type))) {
            return Response.status(Response.Status.BAD_REQUEST).build();
        } else if (!ORGANIZATION_TYPE.equals(type)) {
            return Response.status(Response.Status.NOT_FOUND).build();
        }
```

Reply:
Addressed with https://github.com/wso2/identity-api-server/pull/495/commits/f15e2dac78ced6aee72e580d1fad018ea99574c3